pandas/_libs/tslib.pyx

```diff
@@ -1577,7 +1589,17 @@ cpdef convert_str_to_tsobject(object ts, object tz, object unit,
                     return obj
                 else:
                     # Keep the converter same as PyDateTime's
-                    ts = Timestamp(obj.value, tz=obj.tzinfo)
+                    obj2 = convert_to_tsobject(obj.value, obj.tzinfo,
```

Comment:
this is just a variation on  ``create_datatime_from_ts``

Reply:
I don't follow.